src/probability_theory/probability_space.lean

```diff
@@ -0,0 +1,777 @@
+/-
+Copyright (c) 2021. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Martin Zinkevich (modified for mathlib by Hunter Monroe)
+ -/
+
+import measure_theory.measurable_space
+import measure_theory.measure_space
+
+/-!
+This file defines the basic concepts in probability theory.
+probability_space: a measure_space where the measure has a value of 1.
+event: a set that is measurable (defined based upon the measurable space),
+Pr[E]: the probability of an event.
+-/
+
+open measure_theory measurable_space
+open_locale big_operators classical
+
+namespace probability_theory
+
+/- An event is a measurable set in a measurable space that has a probability measure. Note:
+the definition below does not require that the measurable_space have volume one. -/
+@[derive [has_union,has_emptyc,has_compl, has_sdiff,has_union,has_inter,boolean_algebra]]
+def event (Ω : Type*) [measurable_space Ω] : Type* := {x : set Ω // measurable_set x}
+
+/- A probability space is a measure space with volume one. -/
+class probability_space (α : Type*) extends measure_space α := (volume_univ : volume (set.univ) = 1)
```

Comment:
Mathlib already has `probability_measure`; I'd be inclined to replace all uses of `[probability_space Ω]` with `[measure_space Ω] [probability_measure (volume : measure Ω)]`

Reply:
The standard treatment in the probability literature is to start with a probability triple, consisting of a probability space, its measurable sets, and the probability function. So I do not think we can avoid mentioning probability space. mathlib itself presents measure theory results by defining a measurable_space, and only then the measure. The natural approach would be to follow the same sequence. As a practical matter, in the probability literature, theorems are set up in every possible way: "Given a [probability space OR set of events OR probability measure OR random variables] satisfying ..." so we should allow that flexibility in stating theorems in mathlib. Note: my file needs adjustment to use probability_measure μ rather than the new notation Pr[...]. 

Also, ideally any theorems about events/measures that are just restatements of set/measure theoretic results like (A ∪ B)ᶜ = (Aᶜ ∩ Bᶜ) could ideally be dropped--how can I engineer this? Events are just a subtype of measurable sets for a probability measure, with a deriving . Also, I should scrub away any remaining lemma names like "event and" (eand) as just set intersection. This file could ideally be much shorter than nearly 800 lines. 

To tie together with existing results in mathlib, probability mass functions can be presented as a special case of probability measures with a finite support. Also, several results in probability_theory/integration refer to the expectation operator without defining it, for instance, ∫⁻ a, f a * g a ∂μ rather than E[f * g].  Let me your thoughts.

Reply:
> Also, ideally any theorems about events/measures that are just restatements of set/measure theoretic results like (A ∪ B)ᶜ = (Aᶜ ∩ Bᶜ) could ideally be dropped--how can I engineer this

Just don't state them, the `boolean_algebra` instance already provides all these theorems for you.

Reply:
> The standard treatment in the probability literature is to start with a probability triple, consisting of a probability space, its measurable sets, and the probability function.

Right, these are respectively `Ω : Type*`, `[measure_space Ω]`, and `volume` given `[probability_measure (volume : measure Ω)]`.

> As a practical matter, in the probability literature, theorems are set up in every possible way: "Given a [probability space OR set of events OR probability measure OR random variables] satisfying ..." so we should allow that flexibility in stating theorems in mathlib.

I don't agree with this - having multiple ways to state something means you have to state every theorem in every possible way, which is more work than just picking one or two ways and sticking with them. There is obviously an argument for restating general measure theory results in a way that is specific to probability theory for the sake of ease of understanding for the reader, but such a task would likely live outside mathlib, as a "mathlib adapter for probability users" library.

Reply:
I have deleted probability_space.lean, and just added two lemmas about probability measures to the existing measure_space.lean and fixed the ordering/sections. I will focus on a new branch on random variables, where there is more substance.